src/ExpressionEvaluator/Core/Source/ResultProvider/Formatter.TypeNames.cs

```diff
@@ -92,12 +97,14 @@ protected void AppendQualifiedTypeName(StringBuilder builder, Type type, bool es
         /// <remarks>
         /// Does not call itself or <see cref="AppendQualifiedTypeName"/> (directly).
         /// </remarks>
-        private void AppendQualifiedTypeNameInternal(StringBuilder builder, Type type, bool escapeKeywordIdentifiers)
+        private void AppendQualifiedTypeNameInternal(StringBuilder builder, Type type, ref DynamicFlags dynamicFlags, bool escapeKeywordIdentifiers)
         {
-            if (AppendSpecialTypeName(builder, type, escapeKeywordIdentifiers))
+            var isDynamic = dynamicFlags.GetNextFlag() && type.IsObject();
+            if (AppendSpecialTypeName(builder, type, isDynamic, escapeKeywordIdentifiers))
```

Comment:
It looks like we walk through the flags sequentially. Why can't the current position be an argument rather than a mutable field, as you suggested?